pkg/query-service/app/clickhouseReader/reader.go

```diff
@@ -1551,6 +1551,291 @@ func (r *ClickHouseReader) setTTLTraces(ctx context.Context, orgID string, param
 	return &model.SetTTLResponseItem{Message: "move ttl has been successfully set up"}, nil
 }
 
+func (r *ClickHouseReader) SetCustomRetentionTTL(ctx context.Context, orgID string, params *model.CustomRetentionTTLParams) (*model.CustomRetentionTTLResponse, *model.ApiError) {
+	// Keep only latest 100 transactions/requests
+	r.deleteTtlTransactions(ctx, orgID, 100)
+
+	uuidWithHyphen := uuid.New()
+	uuid := strings.Replace(uuidWithHyphen.String(), "-", "", -1)
```

Comment:
can you check why is this added ? 

if this can't be removed use `valuer.GenerateUUID()` instead of `uuid.New()`

Reply:
The deleteTtlTransactions call and UUID generation appear to be for tracking/cleanup purposes. The UUID is used as a TransactionID to group related operations
It is also being used in `setTTLTraces` function